external/build/youtube-playlist.js

```diff
@@ -0,0 +1,204 @@
+ */
+
+require('dotenv').config();
+
+const fs = require('fs').promises;
+const path = require('path');
+const ms = require('ms');
+const {google} = require('googleapis');
+
+/**
+ * Time interval to determine if the current data set is stale
+ */
+const FETCH_INTERVAL = ms('4h');
+
+/**
+ * Config constants to interact with the YouTube API
+ */
+const API_KEY = process.env.YOUTUBE_API_KEY;
+const PART = ['snippet'];
+const MAX_RESULTS = 50;
+
+/**
+ * Set Google Chrome Developers Youtube channel ID (https://www.youtube.com/@ChromeDevs)
+ */
+const CHANNELS = ['UCnUYZLuoy1rq1aVMwx4aTzw'];
+
+/**
+ * Target directory for the fetched data
+ */
+const targetFile = path.join(__dirname, '../data/youtube-playlist.json');
+const currentTimestamp = Date.now();
+
+const result = {};
+
+const youtube = google.youtube('v3');
+
+/**
+ * Starts the data fetching process form the channel name
+ * specified in the config.
+ */
+
+async function run() {
+  if (!process.env.YOUTUBE_API_KEY) {
+    throw new Error('No `YOUTUBE_API_KEY` environment var for production');
+  }
+
+  /**
+   * Fetch current data set and checks timestamp to see if the data is stale
+   */
+  let dataAge = 0;
+  try {
+    const currentDataRaw = await fs.readFile(targetFile, {encoding: 'utf8'});
+    const currentData = JSON.parse(currentDataRaw);
+
+    dataAge = currentTimestamp - currentData.timestamp;
+  } catch (error) {
+    console.warn(
+      'No existing data found or existing data malformatted. Fetching from YouTube...'
+    );
+  }
+  if (dataAge > FETCH_INTERVAL || !dataAge) {
+    result.timestamp = currentTimestamp;
+
+    result.playlists = [];
+    result.channels = [];
+
+    try {
+      const channelRes = await getChannelData(CHANNELS);
+      result.channels.push(channelRes);
+    } catch (error) {
+      console.error(error);
+      throw new Error('Error fetching the channel data');
+    }
+
+    await Promise.all(
+      CHANNELS.map(async channel => {
+        try {
+          const playlistRes = await getPlaylistData(channel.trim());
+          result.playlists = [...result.playlists, ...playlistRes];
+        } catch (error) {
+          console.error(error);
+          throw new Error('Error fetching the playlist data');
+        }
+      })
+    );
+
+    await fs.writeFile(targetFile, JSON.stringify(result));
+  }
+}
+
+/**
+ * Fetches all the meta data for the playlist and then calls the function
+ * needed for fetching the single videos data
+ * @param {string} id The YouTube channel id
+ * @return {promise} A promise that resolves in the YouTube playlist data
+ * neeeded to display the playlist component
+ */
+
+async function getPlaylistData(id) {
+  const data = [];
+  const response = await youtube?.playlists?.list({
+    part: PART,
+    channelId: id,
+    auth: API_KEY,
+    maxResults: MAX_RESULTS,
+  });
+
+  const playlists = response?.data?.items;
+
+  if (!playlists || playlists.length === 0) {
+    console.warn('No playlist found');
```

Comment:
Similar to above, to make things easier to read and don't require the `else` indentation, instead early return.